src/webgpu/api/validation/encoding/queries/general.spec.ts

```diff
@@ -5,10 +5,100 @@ TODO:
     - queryIndex {in, out of} range for GPUQuerySet
     - GPUQuerySet {valid, invalid}
         - or {undefined}, for occlusionQuerySet
-    - ?
+    - x = {occlusion, pipeline statistics, timestamp} query
 `;
 
+import { params, poptions } from '../../../../../common/framework/params_builder.js';
 import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { kQueryTypes } from '../../../../capability_info.js';
 import { ValidationTest } from '../../validation_test.js';
 
+async function selectDeviceForQueryType(t: ValidationTest, type: GPUQueryType): Promise<void> {
+  const extensions: GPUExtensionName[] = [];
+  if (type === 'pipeline-statistics') {
+    extensions.push('pipeline-statistics-query');
+  } else if (type === 'timestamp') {
+    extensions.push('timestamp-query');
+  }
+
+  await t.selectDeviceOrSkipTestCase({ extensions });
+}
+
+export const enum EncoderType {
+  CommandEncoder = 'CommandEncoder',
+  ComputeEncoder = 'ComputeEncoder',
+  RenderEncoder = 'RenderEncoder',
+}
+
 export const g = makeTestGroup(ValidationTest);
+
+g.test('writeTimestamp')
+  .desc(
+    `
+Tests that write timestamp to all types of query set on all possible encoders:
+- queryIndex {in, out of} range for GPUQuerySet
+- GPUQuerySet {valid, invalid}
+- x= {occlusion, pipeline statistics, timestamp} query
+- x= {compute, render, non-pass} enconder
+  `
+  )
+  .params(
+    params()
+      .combine(
+        poptions('encoderType', [
+          EncoderType.CommandEncoder,
+          EncoderType.ComputeEncoder,
+          EncoderType.RenderEncoder,
+        ] as const)
+      )
+      .combine(poptions('type', kQueryTypes))
+      .combine(poptions('queryIndex', [0, 1, 2]))
+      .unless(({ type, queryIndex }) => type !== 'timestamp' && queryIndex !== 0)
```

Comment:
Can you explain why these cases are skipped in a comment?

Generally we avoid skipping cases in validation tests, if it's not too hard, because it makes it easier to reason about what's being tested (don't have to think about the contents of the `unless`).